Documentation updates

phanimahesh opened the pull request.

spring2maz commented on source/advanced.rst:
```diff
@@ -17,19 +17,41 @@ Shared Subscription
+The topic may contain wildcards (:code:`+` or :code:`#`) in it as per usual rules.
+Group is not allowed to contain the characters :code:`+`,:code:`#`, or :code:`/` in it.
+
+It is recommended to use QoS 2 when using shared subscriptions.
```
No such recommendation that I know of.

phanimahesh commented:
Rebased on top of current develop. Rescinded the qos2 recommendation, and reverted hiding the toc because it is also hiding the sidebar in our theme. There are more updates already written, I'll push those as a separate PR once this is merged.

phanimahesh commented:
@turtleDeng Can this be merged?
